### PR TITLE
New UI for Folder breakdown + Commit workflow for Folder breakdown

### PR DESCRIPTION
**Pull request by hbjORbj**
### Linear
https://linear.app/intuita/issue/INT-820/smart-defaults-for-branching-out-and-creating-pr

https://user-images.githubusercontent.com/32841130/233644720-a9316a62-384f-4903-8aad-90bb82a9bdcb.mov

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -1267,20 +1267,126 @@ export async function activate(context: vscode.ExtensionContext) {
...
+								stagedJobs,
+								createPullRequest: false,
+								createNewBranch: false,
+							},
+							loading: false,
+							error: '',
+						},
+					});
+				} catch (e) {
+					vscode.window.showErrorMessage((e as Error).message);
```
Comment (DmytroHryshyn) [inline src/extension.ts:1388]: `e instanceof Error ? e.message : String(e);`

File: intuita-webview/src/main/TreeItem/index.tsx [hunk 2]
```diff
@@ -14,6 +14,9 @@ type Props = {
 	onClick(): void;
 	depth: number;
 	disabled: boolean;
+	color: string;
+	index: number;
+	isLastChild: boolean;
```
Comment (grzpab) [inline intuita-webview/src/main/TreeItem/index.tsx:19]: `lastChild`

File: intuita-webview/src/main/TreeView/index.tsx [hunk 3]
```diff
@@ -69,19 +69,25 @@ const TreeView = ({ node }: Props) => {
...
 	}: {
 		node: TreeNode;
 		depth: number;
 		open: boolean;
 		setIsOpen: (value: boolean) => void;
 		focusedNodeId: string;
 		setFocusedNodeId: (value: string) => void;
+		color: string;
+		index: number;
+		isLastChild: boolean;
```
Comment (grzpab) [inline intuita-webview/src/main/TreeView/index.tsx:90]: `lastChild`

File: src/extension.ts [hunk 4]
```diff
@@ -1267,20 +1267,126 @@ export async function activate(context: vscode.ExtensionContext) {
...
+
+					const {
+						path,
+						hash,
+						jobHashes: _jobHashes,
+					} = validation.right;
+
+					const jobHashes = _jobHashes.slice() as JobHash[];
+
+					if (!repositoryService) {
```
[the remark this answers is not in the document]
Comment (hbjORbj) [reply src/extension.ts:1296]: addressed

File: src/extension.ts (hunk 1 above)
Comment (hbjORbj) [reply src/extension.ts:1388]: addressed

File: intuita-webview/src/shared/Tree/index.tsx [hunk 5]
```diff
@@ -1,41 +1,71 @@
...
+		color,
+		isLastChild,
 	}: {
+		index: number;
 		node: TreeNode;
 		depth: number;
 		open: boolean;
 		setIsOpen: (value: boolean) => void;
+		color: string;
+		isLastChild: boolean;
```
Comment (grzpab) [inline intuita-webview/src/shared/Tree/index.tsx:25]: `lastChild`

**Review by grzpab — APPROVED** [review]

**Review by DmytroHryshyn — APPROVED** [review]

**Comment by hbjORbj** [comment]
Will open UI tweaks in a separate PR

**Pull request merged**